Index: starlette/tests/test_routing.py
===================================================================
--- starlette.orig/tests/test_routing.py
+++ starlette/tests/test_routing.py
@@ -655,9 +655,11 @@ def test_lifespan_with_on_events(test_cl
         shutdown_called = True
 
     with pytest.deprecated_call(match="The on_startup and on_shutdown parameters are deprecated"):
-        with pytest.warns(
-            UserWarning, match="The `lifespan` parameter cannot be used with `on_startup` or `on_shutdown`."
-        ):
+        # We temporarily do not allow stacking of the pytest contextlib functions as
+        # pytest 7.x doesn't seem to support this
+        #with pytest.warns(
+        #    UserWarning, match="The `lifespan` parameter cannot be used with `on_startup` or `on_shutdown`."
+        #):
             app = Router(on_startup=[run_startup], on_shutdown=[run_shutdown], lifespan=lifespan)
 
             assert not lifespan_called
